src/pages/index.js

```diff
@@ -21,103 +22,264 @@ import {
 } from "../utils/constants.js";
 
-/**************
- * VALIDATION *
- **************/
-const addFormValidator = new FormValidator(
-  validationSettings,
-  addCardEditForm
-);
-addFormValidator.enableValidation();
-
-const editFormValidator = new FormValidator(
-  validationSettings,
-  profileEditForm
-);
-
-editFormValidator.enableValidation();
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/group-12",
+  authToken: "9c860865-e6d3-4014-b437-60037dde85fb",
+});
 
 /*************
- * CARD INFO *
+ * USER INFO *
  *************/
+const userInfo = new UserInfo({
+  userNameSelector,
+  userDescriptionSelector,
+  avatarSelector,
+});
 
-const userInfo = new UserInfo({ userNameSelector, userDescriptionSelector });
+let userId, cardSection;
+
+function renderCard(cardData) {
+  const cardImage = createCard(cardData);
+  cardSection.prependItem(cardImage);
+}
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cardData]) => {
+    userInfo.setUserInfo({
+      title: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+    // userInfo.setAvatarInfo(userData.avatar);
+    userId = userData._id;
+    cardSection = new Section(
+      {
+        data: cardData,
+        renderer: renderCard,
+      },
+      cardListSelector
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 export const cardTemplate = document
   .querySelector("#card-template")
   .content.querySelector(".card");
 
+/********************
+ * POPUP WITH IMAGE *
+ ********************/
 const modalWithImage = new PopupWithImage({
-  popupSelector: imageModalSelector,
+  modalSelector: imageModalSelector,
 });
 
 const modalWithFormUser = new PopupWithForm({
-  popupSelector: profileModalSelector,
-  handleFormSubmit: (cardData) => {
-    userInfo.setUserInfo(cardData);
+  modalSelector: modalChangeProfileAvatarSelector,
+  handleFormSubmit: (data) => {
+    modalWithFormUser.renderLoading(true);
+    api
+      .updateUserProfile({ avatar: data.url })
+      .then((data) => {
+        userInfo.setAvatarInfo(data.avatar);
+        modalWithFormUser.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        modalWithFormUser.renderLoading(false);
+      });
   },
+  loadingText: "Saving...",
+});
+
+console.log('profileModalSelector: ', profileModalSelector)
+
+const modalFormUser = new PopupWithForm({
+  modalSelector: profileModalSelector,
+  handleFormSubmit: (data) => {
+    modalFormUser.renderLoading(true);
+    api
+      .updateUserInfo(data)
+      .then((data) => {
+        userInfo.setUserInfo({
+          title: data.name,
+          description: data.about,
+        });
+        userInfo.setAvatarInfo(data.avatar);
+        modalFormUser.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        modalFormUser.renderLoading(false);
+        if ("edit-modal-form" in formValidators) {
```

Comment:
You should not reset the validation here, reset it before opening the popup (for the corresponding validator). The same for other api calls.